span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -121,33 +97,7 @@ PreProcessedSpan preProcessSpan(Span span) {
 
     String tenantId = maybeTenantId.get();
```

Comment:
@skjindal93 We would want the tenantId to pass on to the PreProcessedSpan class later on in case it is valid. Do we want to leave this tenantId based filtering here as is? Or do we want this as well in SpanDropManager and make it return null in case we do drop span, else the tenantId? Which would be better?

Reply:
We could have ideally, just moved the section 

```
if (this.tenantIdsToExclude.contains(tenantId)) {
  LOG.debug("Dropping span for tenant id : {}", tenantId);
  return Optional.empty();
}
```

outside. And, let the downstream process get the tenant id again, via `getAllowedTenantId`. Since, the span would have already been excluded, there wouldn't be any span at all for the method to be called on